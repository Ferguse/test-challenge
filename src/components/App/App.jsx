import { BrowserRouter } from 'react-router-dom';

import { Router } from 'routes';

import './app.scss';

export const App = () => {
  return (
    <div className="app">
      <BrowserRouter>
        <Router />
      </BrowserRouter>
    </div>
  );
};
