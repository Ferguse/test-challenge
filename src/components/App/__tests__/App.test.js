import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockedStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { App } from '../App';

describe('components [App]', () => {
  const state = {
    wishList: {},
    categories: {
      entities: [],
    },
    current: {},
  };

  const store = configureMockedStore([thunk])(state);

  it('Renders', () => {
    const tree = renderer
      .create(<Provider store={store}><App /></Provider>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
