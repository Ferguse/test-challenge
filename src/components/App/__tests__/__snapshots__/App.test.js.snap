// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components [App] Renders 1`] = `
<div
  className="app"
>
  <div>
    <header
      className="header bg-white"
    >
      <div
        className="container flex justify-between"
      >
        <div
          className="bold logo"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            ELITE SPORT
          </a>
        </div>
        <div>
          <a
            href="/wish-list"
            onClick={[Function]}
          >
            <i
              className="fi fi-rs-heart"
            />
            <span>
               Wish list 
            </span>
          </a>
        </div>
      </div>
    </header>
    <nav
      className="nav"
    >
      <div
        className="container"
      >
        <ul
          className="nav-list flex"
        >
          <li
            className="nav-item"
          >
            <a
              href="/"
              onClick={[Function]}
            >
              <i
                className="fi fi-rs-home nav-icon"
              />
              <span>
                 Home
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              href="/category/bike"
              onClick={[Function]}
            >
              <i
                className="fi fi-rs-bike nav-icon"
              />
              <span>
                 
                bike
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              href="/category/hiking"
              onClick={[Function]}
            >
              <i
                className="fi fi-rs-hiking nav-icon"
              />
              <span>
                 
                hiking
              </span>
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              href="/category/skating"
              onClick={[Function]}
            >
              <i
                className="fi fi-rs-skating nav-icon"
              />
              <span>
                 
                skating
              </span>
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <main
      className="content bg-white"
    >
      <div
        className="container"
      >
        <div
          className="carousel"
        >
          <h2
            className="title"
          >
            <a
              href="/category/bike"
              onClick={[Function]}
            >
              bike
            </a>
            <i
              className="carousel_indicator"
              style={
                {
                  "backgroundColor": "#ffa2a2",
                }
              }
            />
          </h2>
          <div
            className="slick-slider slick-initialized"
          >
            
            <div
              className="slick-list"
            >
              <div
                className="slick-track"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  {
                    "WebkitTransform": "translate3d(0px, 0px, 0px)",
                    "WebkitTransition": "",
                    "msTransform": "translateX(0px)",
                    "opacity": 1,
                    "transform": "translate3d(0px, 0px, 0px)",
                    "transition": "",
                  }
                }
              />
            </div>
            
            
          </div>
        </div>
        <div
          className="carousel"
        >
          <h2
            className="title"
          >
            <a
              href="/category/hiking"
              onClick={[Function]}
            >
              hiking
            </a>
            <i
              className="carousel_indicator"
              style={
                {
                  "backgroundColor": "#a7fda7",
                }
              }
            />
          </h2>
          <div
            className="slick-slider slick-initialized"
          >
            
            <div
              className="slick-list"
            >
              <div
                className="slick-track"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  {
                    "WebkitTransform": "translate3d(0px, 0px, 0px)",
                    "WebkitTransition": "",
                    "msTransform": "translateX(0px)",
                    "opacity": 1,
                    "transform": "translate3d(0px, 0px, 0px)",
                    "transition": "",
                  }
                }
              />
            </div>
            
            
          </div>
        </div>
        <div
          className="carousel"
        >
          <h2
            className="title"
          >
            <a
              href="/category/skating"
              onClick={[Function]}
            >
              skating
            </a>
            <i
              className="carousel_indicator"
              style={
                {
                  "backgroundColor": "#a9a9fb",
                }
              }
            />
          </h2>
          <div
            className="slick-slider slick-initialized"
          >
            
            <div
              className="slick-list"
            >
              <div
                className="slick-track"
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                onMouseOver={[Function]}
                style={
                  {
                    "WebkitTransform": "translate3d(0px, 0px, 0px)",
                    "WebkitTransition": "",
                    "msTransform": "translateX(0px)",
                    "opacity": 1,
                    "transform": "translate3d(0px, 0px, 0px)",
                    "transition": "",
                  }
                }
              />
            </div>
            
            
          </div>
        </div>
      </div>
    </main>
    <footer>
      <div
        className="container"
      >
        <p>
          Ipsum.
        </p>
        <p>
          Dolor nostrud est cupidatat aute magna cupidatat do consequat magna nisi voluptate. Sit aliquip exercitation nulla proident sint sint amet exercitation mollit anim aliquip velit. Do adipisicing Lorem laboris enim velit sit nulla sint nisi excepteur consequat id ad exercitation. Irure labore incididunt.
        </p>
        <p>
          Et ad et incididunt tempor.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
